File: examples/php/client_http.php
<?php
$base = "http://localhost:8080";
$db = "demo.db";

$payload = json_encode(["db"=>$db, "sql"=>"SELECT * FROM users LIMIT 10;"]);
$ch = curl_init($base."/exec");
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, ["Content-Type: application/json"]);
curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);

$response = curl_exec($ch);
if ($response === false) { echo "curl error: ".curl_error($ch).PHP_EOL; exit(1); }
curl_close($ch);

echo $response.PHP_EOL;

File: examples/python/client_http.py
import json
import urllib.request

BASE = "http://localhost:8080"
DB = "demo.db"  # en modo -dir, debe existir dentro de ./dbs

def post(path, payload):
    req = urllib.request.Request(
        BASE + path,
        data=json.dumps(payload).encode("utf-8"),
        headers={"Content-Type": "application/json"},
        method="POST",
    )
    with urllib.request.urlopen(req) as resp:
        return resp.read().decode("utf-8")

print(post("/exec", {"db": DB, "sql": "SELECT * FROM users LIMIT 10;"}))
